### PR TITLE
add InjectIntoImageCanvas

### PR DESCRIPTION
**Pull request by meihuisu**
combine scalebar with the underlying viewer image

### CODE REVIEW

File: openseadragon-scalebar.js [hunk 1]
```diff
@@ -371,6 +371,22 @@
             var vCenter = canvas.height / 2;
             context.fillText(this.divElt.textContent, hCenter, vCenter);
             return canvas;
+        },
+        /**
+         * Inject scalebar into a canvas with viewport image.
+         * @returns {Element} A canvas containing the scalebar representation
+         */
+        injectIntoImageCanvas: function() {
```
Comment (avandecreme) [inline openseadragon-scalebar.js]: I think that `getImageWithScalebarAsCanvas` would be a better name. Would you mind changing it?

For the documentation, I suggest:

> Get a copy of the current OpenSeadragon canvas with the scalebar.
> @returns {Element} A canvas containing a copy of the current OpenSeadragon canvas with the scalebar

File: openseadragon-scalebar.js [hunk 2]
```diff
@@ -371,6 +371,22 @@
...
+            var imgCanvas = this.viewer.drawer.canvas;
+            var newCanvas = document.createElement("canvas");
+            newCanvas.width = imgCanvas.width;
+            newCanvas.height = imgCanvas.height;
+            var newCtx = newCanvas.getContext('2d');
```
Comment (avandecreme) [inline openseadragon-scalebar.js]: Please use double quotes for consistency.

File: openseadragon-scalebar.js [hunk 3]
```diff
@@ -371,6 +371,22 @@
...
+            newCtx.drawImage(imgCanvas,0,0);
```
Comment (avandecreme) [inline openseadragon-scalebar.js]: Please add spaces after commas.

File: openseadragon-scalebar.js [hunk 4]
```diff
@@ -371,6 +371,22 @@
...
+            var scalebarCanvas = this.getAsCanvas();
+            var location = this.getScalebarLocation();
+            newCtx.drawImage(scalebarCanvas,location.x,location.y);
```
Comment (avandecreme) [inline openseadragon-scalebar.js]: Please add spaces after commas.

**Comment by avandecreme** [comment]
Thanks! I made some minor comments. I will merge when you have fixed them and I get push access to this repository (might take some time).

**Comment by avandecreme** [comment]
Thank you very much!

**Comment by meihuisu** [comment]
@avandecreme,  I was just cloning another copy and noticed that the last set of changes  according to your suggestions were reverted back to original.

**Comment by avandecreme** [comment]
It looks correct here: https://github.com/usnistgov/OpenSeadragonScalebar/blob/master/openseadragon-scalebar.js#L375-L391
I did a merge squashing, I wonder if that could have messed up your local history? What if you do a fresh clone?

**Comment by meihuisu** [comment]
It is suppose to be 

```
    /**
     * Inject scalebar into a canvas with viewport image.
     * @returns {Element} A canvas containing the scalebar representation
     */
    injectIntoImageCanvas: function() {
```

as you have asked.

----- Original Message -----

**Comment by avandecreme** [comment]
I asked you to change it from `injectIntoImageCanvas` to `getImageWithScalebarAsCanvas`. It looks good to me.

**Comment by meihuisu** [comment]
Thanks. I thought I might have done something accidentally.

----- Original Message -----

**Pull request merged**